pkg/chain/bitcoin/derive_address_test.go

```diff
@@ -171,158 +180,105 @@ func ErrorContains(err error, expected string) bool {
-			dis, err := NewDerivationIndexStorage(dir)
-			if err != nil {
-				t.Fatal(err)
-			}
-			for _, usedIndex := range testData.usedIndexes {
-				dis.save(testData.beneficiaryAddress, usedIndex)
-			}
-
-			handle := newMockBitcoinHandle()
-
-			resolvedAddress, err := ResolveAddress(
-				testData.beneficiaryAddress,
-				dis,
-				testData.chainParams,
-				handle,
-			)
+			err := ValidateAddress(testData.beneficiaryAddress, testData.chainParams)
 			if err != nil {
-				t.Fatal(err)
-			}
-			if resolvedAddress != testData.expectedAddress {
-				t.Errorf(
-					"the resolved address does not match\nexpected: %s\nactual:   %s",
-					testData.expectedAddress,
-					resolvedAddress,
-				)
+				t.Error(err)
 			}
 		})
 	}
 }
 
-var resolveAddressExpectedFailureData = map[string]struct {
-	extendedAddress string
-	chainParams     *chaincfg.Params
-	failure         string
-}{
-	"WIF": {
-		"5Hwgr3u458GLafKBgxtssHSPqJnYoGrSzgQsPwLFhLNYskDPyyA",
-		&chaincfg.MainNetParams,
-		"the provided serialized extended key length is invalid",
-	},
-	"empty string": {
-		"",
-		&chaincfg.MainNetParams,
-		"insufficient length for public key",
-	},
-	"BIP32 private key": {
-		"xprv9s21ZrQH143K24Mfq5zL5MhWK9hUhhGbd45hLXo2Pq2oqzMMo63oStZzF93Y5wvzdUayhgkkFoicQZcP3y52uPPxFnfoLZB21Teqt1VvEHx",
-		&chaincfg.MainNetParams,
-		"unusable seed",
-	},
-	"complete nonsense": {
-		"lorem ipsum dolor sit amet, consec",
-		&chaincfg.MainNetParams,
-		"the provided serialized extended key length is invalid",
-	},
-}
-
-func TestResolveBeneficiaryAddress_ExpectedFailure(t *testing.T) {
-	for testName, testData := range resolveAddressExpectedFailureData {
+func TestValidateAddress_ExpectedFailures(t *testing.T) {
+	var testData = map[string]struct {
+		beneficiaryAddress string
+		chainParams        *chaincfg.Params
+		err                string
+	}{
+		"nonsense address": {
+			"banana123",
+			&chaincfg.MainNetParams,
+			"[banana123] is not a valid btc address using chain [mainnet]: [error parsing extended public key: [the provided serialized extended key length is invalid]]",
+		},
+		"empty string": {
+			"",
+			&chaincfg.RegressionNetParams,
+			"[] is not a valid btc address using chain [regtest]: [error parsing extended public key: [the provided serialized extended key length is invalid]]",
+		},
+		"Mainnet private key": {
+			"5Hwgr3u458GLafKBgxtssHSPqJnYoGrSzgQsPwLFhLNYskDPyyA",
+			&chaincfg.MainNetParams,
+			"[5Hwgr3u458GLafKBgxtssHSPqJnYoGrSzgQsPwLFhLNYskDPyyA] is not a valid btc address using chain [mainnet]: [error parsing extended public key: [the provided serialized extended key length is invalid]]",
+		},
+		"testnet private key": {
+			"92Pg46rUhgTT7romnV7iGW6W1gbGdeezqdbJCzShkCsYNzyyNcc",
+			&chaincfg.TestNet3Params,
+			"[92Pg46rUhgTT7romnV7iGW6W1gbGdeezqdbJCzShkCsYNzyyNcc] is not a valid btc address using chain [testnet3]: [error parsing extended public key: [the provided serialized extended key length is invalid]]",
+		},
+	}
```

Comment:
Can we add a case where the address is testnet's address and chain params are for the mainnet?